### ISSUE TITLE
Migrate to JUCE6 and convert .jucer to CMakeLists.txt

### ISSUE THREAD

**Issue by atsushieno** [atsushieno/aap-juce#1]
Things have been improved from "build manually copied sources with modification" to "copy from submodule, apply tiny patch, alter .jucer and build" now. As a next step, if the .jucer files can be automatically converted from original sources, that would be great.

Most of the changes are already described on README.md, and it wouldn't be too hard to create some sort of converter script that manipulates the .jucer XML files.

**Comment by atsushieno** [comment]
We could wait for JUCE6 and migrate to CMake.

**Comment by atsushieno** [comment]
Before starting the actual migration, we have to make sure that existing audio plugins build with JUCE6. Migration to JUCE6 without apps does not make a lot of sense.

**Comment by atsushieno** [comment]
Thoughts on juce6 migration now that it is released:

- Migration to juce6 is not very problematic.
- Migration to CMake can be done later.
- Migration to juce6 brings on some good: improved Oboe support and some Android project generation bits.
- Migration to juce6 blocks juce_emscripten updates. juce_emscripten needs some further changes than just merging.
  - We can abandon juce_emscripten and go back to JUCE6 until Dreamtonics comes back to the stack (which hasn't happened for months).

We can work on it now.

**Comment by atsushieno** [comment]
some blocker: https://github.com/juce-framework/JUCE/issues/753  (workaround: just make some local changes in JUCE code)

**Comment by atsushieno** [comment]
juce6 branch is now merged into main.
